app/forms/webauthn_setup_form.rb

```diff
@@ -43,6 +43,22 @@ def platform_authenticator?
     !!@platform_authenticator
   end
 
+  def error_message
+    if errors.where(:name).present?
+      if errors.first.options[:type]
+        return errors.first.message
+      end
+    end
+    if @platform_authenticator
+      I18n.t('errors.webauthn_platform_setup.general_error')
+    else
+      I18n.t(
+        'errors.webauthn_setup.general_error_html',
+        link_html: I18n.t('errors.webauthn_setup.additional_methods_link'),
+      )
+    end
```

Comment:
I'd consider removing the early return and combining these?
```suggestion
    if errors.where(:name).present? && errors.first.options[:type]
      errors.first.message
    elsif @platform_authenticator
      I18n.t('errors.webauthn_platform_setup.general_error')
    else
      I18n.t(
        'errors.webauthn_setup.general_error_html',
        link_html: I18n.t('errors.webauthn_setup.additional_methods_link'),
      )
    end
```